.github/workflows/build.yml

```
@@ -30,17 +34,82 @@ jobs:
             machine_arch: 'S/390'
     steps:
       - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      # The checkout action in previous step overwrites annonated tag
+      # with lightweight tags breaking ``git describe``
+      # See https://github.com/actions/checkout/issues/882
+      # and https://github.com/actions/checkout/issues/290
+      # The following step is a workaround to restore the latest tag
+      # to it's original state.
+      - name: Restore (annonated) tag
+        run: git fetch --force origin ${GITHUB_REF}:${GITHUB_REF}
+        if: ${{ github.ref_type == 'tag' }}
 
       - uses: actions/setup-go@v3
         with:
           go-version-file: 'go.mod'
 
       - name: Build Chisel for linux/${{ matrix.arch }}
-        run: GOARCH=${{ matrix.arch }} go build ./cmd/chisel/
+        id: build
+        env:
+          GOOS: "linux"
+          GOARCH: ${{ matrix.arch }}
+          CGO_ENABLED: "0"
+        run: |
+          echo "Generating version file"
+          go generate ./cmd/
+
+          echo "Building for $GOOS $GOARCH"
+          go build -trimpath -ldflags='-s -w' ./cmd/chisel
+
+          # Get version via "chisel version" to ensure it matches that exactly
+          CHISEL_VERSION=$(GOOS=linux GOARCH=amd64 go run ./cmd/chisel version)
+          echo "Version: $CHISEL_VERSION"
+
+          # Version should not be "unknown"
+          [ "$CHISEL_VERSION" != "unknown" ] || exit 1
+
+          # Share variables with subsequent steps
+          echo "CHISEL_VERSION=${CHISEL_VERSION}" >>$GITHUB_OUTPUT
 
       - name: Test if is executable
         run: test -x ./chisel
 
       - name: Test if binary has the right machine architecture
         run: |
           [ "$(readelf -h chisel | grep 'Machine:' | awk -F' ' '{print $NF}')" == "${{ matrix.machine_arch }}" ]
+
+      - name: Create archive
+        id: archive
+        env:
+          GOOS: "linux"
+          GOARCH: ${{ matrix.arch }}
+          CHISEL_VERSION: ${{ steps.build.outputs.CHISEL_VERSION }}
+        run: |
+          ARCHIVE_FILE=chisel_${CHISEL_VERSION}_${GOOS}_${GOARCH}.tar.gz
+          echo "Creating archive $ARCHIVE_FILE"
+
+          mkdir -p dist/build
+          cp chisel LICENSE README.md dist/build
+          find dist/build -printf "%P\n" | tar -czf dist/$ARCHIVE_FILE --no-recursion -C dist/build -T -
```

Is there a reason we're doing this complex rather than something like:

```suggestion
          tar -czf dist/$ARCHIVE_FILE -C dist/build .
```

---

`tar -czf dist/$ARCHIVE_FILE -C dist/build .` creates a `.` directory inside the tarball. The files in the targeted directory are placed in that `.` directory. This is an example list of contents of a tarball created by that command:
```
drwxrwxr-x ubuntu/ubuntu       0 2023-08-10 12:34 ./
-rw-rw-r-- ubuntu/ubuntu       0 2023-08-10 12:34 ./bar
-rw-rw-r-- ubuntu/ubuntu       0 2023-08-10 12:30 ./foo
```

`find dist/build -printf "%P\n" | tar -czf dist/$ARCHIVE_FILE --no-recursion -C dist/build -T -` does not include the `.` directory and includes the files in the target directory at the top-level inside the tarball.
```
-rw-rw-r-- ubuntu/ubuntu       0 2023-08-10 12:34 bar
-rw-rw-r-- ubuntu/ubuntu       0 2023-08-10 12:30 foo
```

Let me know if you prefer it to include the `.` directory.

---

You're right, and now I see that this is also broken in Pebble ;-)